src/v/kafka/client/client.cc

```diff
@@ -166,11 +166,12 @@ ss::future<> client::update_metadata(wait_or_start::tag) {
                       seeds.emplace_back(b.host, b.port);
                   }
                   std::swap(_seeds, seeds);
-
                   return apply(std::move(res));
               })
               .finally([]() { vlog(kclog.trace, "updated metadata"); });
-        });
+        })
+          .handle_exception_type(
+            [this](const broker_error) { return connect(); });
```

Comment:
const-ref?

Reply:
what's a `broker_error` seems super generic.

Reply:
It's an exception that carries broker information, usually due to a disconnection, or sometimes if there are no brokers.